fieldsolver/ldz_magnetic_field.cpp

```diff
@@ -191,8 +191,21 @@ void propagateSysBoundaryMagneticField(
       bGrid = perBDt2Grid.get(i,j,k);
    }
    cuint sysBoundaryFlag = technicalGrid.get(i,j,k)->sysBoundaryFlag;
+   cuint sysBoundaryLayer = technicalGrid.get(i,j,k)->sysBoundaryLayer;
+   
    for (uint component = 0; component < 3; component++) {
-      bGrid->at(fsgrids::bfield::PERBX + component) = sysBoundaries.getSysBoundary(sysBoundaryFlag)->fieldSolverBoundaryCondMagneticField(perBGrid, perBDt2Grid, EGrid, EDt2Grid, technicalGrid, i, j, k, dt, RKCase, component);
+      if (sysBoundaryFlag != 1) {
```

Comment:
This should probably be NOT_SYSTBOUNDARY instead of a hardcoded 1. (I had to look up which it was)

Reply:
Who doesn't know that. :) ;)

Reply:
NO this is a bug! Should be layer!

Reply:
Good catch, that reduced the diffs I am seeing.